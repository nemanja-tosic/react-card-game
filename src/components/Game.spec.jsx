// @flow
import React from 'react';
import { shallow } from 'enzyme';

import { TestGame } from '../test/support/test-game';
import GameComponent from './Game';
import TalonComponent from './Talon';

describe('Game', () => {
  let component;
  let onCardClickedSpy;
  let playerComponent;
  let talonComponent;
  let game;

  beforeEach(() => {
    game = TestGame.create();
    onCardClickedSpy = jest.fn();
    component = shallow(
      <GameComponent game={game} onCardClicked={onCardClickedSpy}/>,
      {lifecycleExperimental: true}
    );
    talonComponent = component.find(TalonComponent);
    playerComponent = component.find('PlayerComponent');
  });

  it('should render correctly', () => {
    expect(component.getNode()).toMatchSnapshot();
  });

  describe('when the game updates', () => {
    let cardDomNode;

    beforeEach(() => {
      cardDomNode = {
        getBoundingClientRect: jest.fn()
          .mockReturnValueOnce({left: 0, top: 0, width: 100, height: 100})
          .mockReturnValueOnce({left: 20, top: 30, width: 0, height: 0}),
        animate: jest.fn(),
      };
    });

    it('should FLIP animate the cards', () => {
      const playedCard = game.players[0].cards.splice(0, 1)[0];
      game.talon = [playedCard];
      playerComponent.prop('cardRef')('JS', cardDomNode);
      talonComponent.prop('cardRef')('JS', cardDomNode);

      component.setProps({game});

      expect(cardDomNode.animate).toHaveBeenCalledWith({
        transform: ['translate(-20px, -30px) scale(1000, 1000)', 'translate(0, 0) scale(1, 1)'],
      }, expect.anything());
    });
  });
});
