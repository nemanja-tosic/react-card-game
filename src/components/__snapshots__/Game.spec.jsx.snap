// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render correctly 1`] = `
<div
  className="Game"
>
  <div
    className="Game__talon"
  >
    <Talon
      cardRef={[Function]}
      cards={
        Array [
          Array [
            Card {
              "code": "JS",
              "imageUrl": "https://deckofcardsapi.com/static/img/JS.png",
            },
          ],
        ]
      }
    />
  </div>
  <div
    className="Game__player Game__player--0"
  >
    <PlayerComponent
      cardRef={[Function]}
      onCardClicked={[Function]}
      player={
        Player {
          "cards": Array [
            Card {
              "code": "0C",
              "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
            },
            Card {
              "code": "2C",
              "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
            },
            Card {
              "code": "3C",
              "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
            },
            Card {
              "code": "4C",
              "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
            },
            Card {
              "code": "5C",
              "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
            },
          ],
          "id": 1,
          "isActive": true,
          "isUserControlled": true,
          "name": "Player",
          "score": 0,
          "takenCards": Array [],
        }
      }
      slot={0}
    />
  </div>
</div>
`;
